### PR TITLE
Metatraffic topic and endpoints [11706]

### PR DESCRIPTION
**Pull request by SerranoePros**
Draft until blackbox tests are implemented.

### CODE REVIEW

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
Only one comment that deeply affects this PR. The tests should also be adapted taking into account this change.

File: src/cpp/subscriber/StatisticsParticipantListener.cpp [hunk 1]
```diff
@@ -341,6 +346,33 @@ void StatisticsParticipantListener::on_participant_discovery(
...
+                {
+                    metatraffic_endpoint = endpoint.second;
+                }
+            }
+            if (nullptr == metatraffic_endpoint)
+            {
+                logError(STATISTICS_BACKEND, "Participant " << to_string(
+                            info.info.m_guid) + " without meta-traffic endpoint")
+            }
+            else
```
Comment (MiguelCompany) [inline src/cpp/subscriber/StatisticsParticipantListener.cpp:366]: Could we keep a reference to the builtin endpoint inside the participant?
This way we could just get the reference here.

File: src/cpp/subscriber/StatisticsParticipantListener.cpp [hunk 2]
```diff
@@ -373,12 +405,109 @@ void StatisticsParticipantListener::on_participant_discovery(
...
+                // Check if is the metatraffic topic already exists
+                std::vector<std::pair<EntityId, EntityId>> metatraffic_topic_ids =
+                        database_->get_entities_by_name(EntityKind::TOPIC, metatraffic_prefix + "TOPIC");
+
+                for (const auto& topic_id : metatraffic_topic_ids)
+                {
+                    if (topic_id.first == domain_id_)
+                    {
+                        metatraffic_topic = std::const_pointer_cast<database::Topic>(
+                            std::static_pointer_cast<const database::Topic>(database_->get_entity(topic_id.second)));
```
Comment (MiguelCompany) [inline src/cpp/subscriber/StatisticsParticipantListener.cpp:427]:
```suggestion
                        metatraffic_topic = std::const_pointer_cast<database::Topic>(
                            std::static_pointer_cast<const database::Topic>(database_->get_entity(topic_id.second)));
                        break;
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Pull request merged**